### PR TITLE
Allow to set a code editor for the custom step

### PR DESCRIPTION
**Pull request by rhuille**
Allow to set a custom code editor for to edit a custom step

**Before:**
	- the CustomStep use the "TextareaWidget" which is a wrapper arround a textarea

**After:**
	- the CustomStep use the "CodeEditor" object exposed by the code-editor script
	- CodeEditor object is by default "DefaultCodeEditor" a vue component (it's a wrapper arround a textarea)
	- The CodeEditor object can be changed by the "setCodeEditor" function exposed in the main script
	- We provide in the playground app an example of CodeEditor Customisation
	- remove the TextAreaWidget

### CODE REVIEW

File: src/components/code-editor.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/*
+  This module handle the CodeEditor used in the CodeEditorWidget
```
Comment (adimascio) [inline src/components/code-editor.ts]:
```suggestion
  This module handles the CodeEditor used in the CodeEditorWidget
```

File: src/components/code-editor.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  The default CodeEditor is a native textarea.
+
+  You can set an other Code Editor passing to `setCodeEditor`
```
Comment (adimascio) [inline src/components/code-editor.ts]:
```suggestion
  You can set another Code Editor passing to `setCodeEditor`
```

File: src/components/stepforms/widgets/DefaultCodeEditor.vue [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+export default class DefaultCodeEditor extends Vue {
+  @Prop({
+    type: String,
+    required: true,
+  })
+  value!: string;
+
+  @Prop({
+    type: String,
+    default: () => '',
```
Comment (adimascio) [inline src/components/stepforms/widgets/DefaultCodeEditor.vue]:
```suggestion
    default: '',
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,11 @@
 
 ## UNRELEASED -
 
+### Added
+
+- expose `setCodeEditor` to set a code editor in the custom step
+- custom step which allow user to edit manually a step
```
Comment (adimascio) [inline CHANGELOG.md]:
```suggestion
- custom step which allows users to edit manually a step
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,11 @@
 
 ## UNRELEASED -
 
+### Added
+
+- expose `setCodeEditor` to set a code editor in the custom step
```
Comment (adimascio) [inline CHANGELOG.md]:
```suggestion
- expose `setCodeEditor` to set a custom code editor component for  custom steps
```

File: docs/_docs/tech/custom-code-editor.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
+---
+title: Custom code editor
+permalink: /docs/tech
```
Comment (adimascio) [inline docs/_docs/tech/custom-code-editor.md]:
```suggestion
permalink: /docs/code-editor/
```

and add the corresponding entry in `_data/docs.yml`

**Comment by arimbr** [comment]
# [Codecov](https://codecov.io/gh/ToucanToco/weaverbird/pull/419?src=pr&el=h1) Report
> Merging [#419](https://codecov.io/gh/ToucanToco/weaverbird/pull/419?src=pr&el=desc) into [master](https://codecov.io/gh/ToucanToco/weaverbird/commit/ad28855351072e22dc5ff62fc2c673eae0a89580&el=desc) will **decrease** coverage by `0.08%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/ToucanToco/weaverbird/pull/419?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master     #419      +/-   ##
==========================================
- Coverage   96.38%   96.29%   -0.09%     
==========================================
  Files         111      111              
  Lines        2487     2431      -56     
  Branches      362      346      -16     
==========================================
- Hits         2397     2341      -56     
  Misses         90       90              
```


| [Impacted Files](https://codecov.io/gh/ToucanToco/weaverbird/pull/419?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [src/components/stepforms/ToDateStepForm.vue](https://codecov.io/gh/ToucanToco/weaverbird/pull/419/diff?src=pr&el=tree#diff-c3JjL2NvbXBvbmVudHMvc3RlcGZvcm1zL1RvRGF0ZVN0ZXBGb3JtLnZ1ZQ==) | `92.85% <0.00%> (-1.27%)` | :arrow_down: |
| [...onents/stepforms/widgets/FilterSimpleCondition.vue](https://codecov.io/gh/ToucanToco/weaverbird/pull/419/diff?src=pr&el=tree#diff-c3JjL2NvbXBvbmVudHMvc3RlcGZvcm1zL3dpZGdldHMvRmlsdGVyU2ltcGxlQ29uZGl0aW9uLnZ1ZQ==) | `94.28% <0.00%> (-0.46%)` | :arrow_down: |
| [src/components/ActionToolbarButton.vue](https://codecov.io/gh/ToucanToco/weaverbird/pull/419/diff?src=pr&el=tree#diff-c3JjL2NvbXBvbmVudHMvQWN0aW9uVG9vbGJhckJ1dHRvbi52dWU=) | `82.22% <0.00%> (-0.39%)` | :arrow_down: |
| [src/components/QueryBuilder.vue](https://codecov.io/gh/ToucanToco/weaverbird/pull/419/diff?src=pr&el=tree#diff-c3JjL2NvbXBvbmVudHMvUXVlcnlCdWlsZGVyLnZ1ZQ==) | `89.74% <0.00%> (-0.26%)` | :arrow_down: |
| [src/store/mutations.ts](https://codecov.io/gh/ToucanToco/weaverbird/pull/419/diff?src=pr&el=tree#diff-c3JjL3N0b3JlL211dGF0aW9ucy50cw==) | `95.23% <0.00%> (-0.15%)` | :arrow_down: |
| [src/lib/translators/mongo.ts](https://codecov.io/gh/ToucanToco/weaverbird/pull/419/diff?src=pr&el=tree#diff-c3JjL2xpYi90cmFuc2xhdG9ycy9tb25nby50cw==) | `95.45% <0.00%> (-0.15%)` | :arrow_down: |
| [src/lib/labeller.ts](https://codecov.io/gh/ToucanToco/weaverbird/pull/419/diff?src=pr&el=tree#diff-c3JjL2xpYi9sYWJlbGxlci50cw==) | `98.13% <0.00%> (-0.10%)` | :arrow_down: |
| [src/lib/translators/base.ts](https://codecov.io/gh/ToucanToco/weaverbird/pull/419/diff?src=pr&el=tree#diff-c3JjL2xpYi90cmFuc2xhdG9ycy9iYXNlLnRz) | `98.50% <0.00%> (-0.03%)` | :arrow_down: |
| [src/components/stepforms/StepForm.vue](https://codecov.io/gh/ToucanToco/weaverbird/pull/419/diff?src=pr&el=tree#diff-c3JjL2NvbXBvbmVudHMvc3RlcGZvcm1zL1N0ZXBGb3JtLnZ1ZQ==) | `98.68% <0.00%> (-0.02%)` | :arrow_down: |
| [src/lib/steps.ts](https://codecov.io/gh/ToucanToco/weaverbird/pull/419/diff?src=pr&el=tree#diff-c3JjL2xpYi9zdGVwcy50cw==) | `100.00% <0.00%> (ø)` | :arrow_up: |
| ... and [11 more](https://codecov.io/gh/ToucanToco/weaverbird/pull/419/diff?src=pr&el=tree-more) | |

------

[Continue to review full report at Codecov](https://codecov.io/gh/ToucanToco/weaverbird/pull/419?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/ToucanToco/weaverbird/pull/419?src=pr&el=footer). Last update [ad28855...db5db6c](https://codecov.io/gh/ToucanToco/weaverbird/pull/419?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Comment by arimbr** [comment]
# [Codecov](https://codecov.io/gh/ToucanToco/weaverbird/pull/419?src=pr&el=h1) Report
> Merging [#419](https://codecov.io/gh/ToucanToco/weaverbird/pull/419?src=pr&el=desc) into [master](https://codecov.io/gh/ToucanToco/weaverbird/commit/ad28855351072e22dc5ff62fc2c673eae0a89580&el=desc) will **decrease** coverage by `0.04%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/ToucanToco/weaverbird/pull/419?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master     #419      +/-   ##
==========================================
- Coverage   96.38%   96.33%   -0.05%     
==========================================
  Files         111      111              
  Lines        2487     2430      -57     
  Branches      362      346      -16     
==========================================
- Hits         2397     2341      -56     
+ Misses         90       89       -1     
```


| [Impacted Files](https://codecov.io/gh/ToucanToco/weaverbird/pull/419?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [src/components/stepforms/ToDateStepForm.vue](https://codecov.io/gh/ToucanToco/weaverbird/pull/419/diff?src=pr&el=tree#diff-c3JjL2NvbXBvbmVudHMvc3RlcGZvcm1zL1RvRGF0ZVN0ZXBGb3JtLnZ1ZQ==) | `92.85% <0.00%> (-1.27%)` | :arrow_down: |
| [...onents/stepforms/widgets/FilterSimpleCondition.vue](https://codecov.io/gh/ToucanToco/weaverbird/pull/419/diff?src=pr&el=tree#diff-c3JjL2NvbXBvbmVudHMvc3RlcGZvcm1zL3dpZGdldHMvRmlsdGVyU2ltcGxlQ29uZGl0aW9uLnZ1ZQ==) | `94.28% <0.00%> (-0.46%)` | :arrow_down: |
| [src/components/ActionToolbarButton.vue](https://codecov.io/gh/ToucanToco/weaverbird/pull/419/diff?src=pr&el=tree#diff-c3JjL2NvbXBvbmVudHMvQWN0aW9uVG9vbGJhckJ1dHRvbi52dWU=) | `82.22% <0.00%> (-0.39%)` | :arrow_down: |
| [src/components/QueryBuilder.vue](https://codecov.io/gh/ToucanToco/weaverbird/pull/419/diff?src=pr&el=tree#diff-c3JjL2NvbXBvbmVudHMvUXVlcnlCdWlsZGVyLnZ1ZQ==) | `89.74% <0.00%> (-0.26%)` | :arrow_down: |
| [src/store/mutations.ts](https://codecov.io/gh/ToucanToco/weaverbird/pull/419/diff?src=pr&el=tree#diff-c3JjL3N0b3JlL211dGF0aW9ucy50cw==) | `95.23% <0.00%> (-0.15%)` | :arrow_down: |
| [src/lib/translators/mongo.ts](https://codecov.io/gh/ToucanToco/weaverbird/pull/419/diff?src=pr&el=tree#diff-c3JjL2xpYi90cmFuc2xhdG9ycy9tb25nby50cw==) | `95.45% <0.00%> (-0.15%)` | :arrow_down: |
| [src/lib/labeller.ts](https://codecov.io/gh/ToucanToco/weaverbird/pull/419/diff?src=pr&el=tree#diff-c3JjL2xpYi9sYWJlbGxlci50cw==) | `98.13% <0.00%> (-0.10%)` | :arrow_down: |
| [src/lib/translators/base.ts](https://codecov.io/gh/ToucanToco/weaverbird/pull/419/diff?src=pr&el=tree#diff-c3JjL2xpYi90cmFuc2xhdG9ycy9iYXNlLnRz) | `98.50% <0.00%> (-0.03%)` | :arrow_down: |
| [src/components/stepforms/StepForm.vue](https://codecov.io/gh/ToucanToco/weaverbird/pull/419/diff?src=pr&el=tree#diff-c3JjL2NvbXBvbmVudHMvc3RlcGZvcm1zL1N0ZXBGb3JtLnZ1ZQ==) | `98.68% <0.00%> (-0.02%)` | :arrow_down: |
| [src/lib/steps.ts](https://codecov.io/gh/ToucanToco/weaverbird/pull/419/diff?src=pr&el=tree#diff-c3JjL2xpYi9zdGVwcy50cw==) | `100.00% <0.00%> (ø)` | :arrow_up: |
| ... and [11 more](https://codecov.io/gh/ToucanToco/weaverbird/pull/419/diff?src=pr&el=tree-more) | |

------

[Continue to review full report at Codecov](https://codecov.io/gh/ToucanToco/weaverbird/pull/419?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/ToucanToco/weaverbird/pull/419?src=pr&el=footer). Last update [ad28855...db5db6c](https://codecov.io/gh/ToucanToco/weaverbird/pull/419?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Pull request merged**